src/compiler/parser.ts

```diff
@@ -4517,6 +4523,12 @@ module ts {
         }
 
         function parseClassDeclaration(fullStart: number, modifiers: ModifiersArray): ClassDeclaration {
+            // In ES6 specification, All parts of a ClassDeclaration or a ClassExpression are strict mode code
+            if (languageVersion >= ScriptTarget.ES6) {

```

Comment:
That will be a breaking change. I have tried running on that mode on RWC and our compiler. Both do have non-strict-mode code in classDeclaration. So, I want to push this version in first. Then, decide if we want to extend this to all classDeclaration later.